source/misc/fish_startup.in

```diff
@@ -1,71 +1,70 @@
-if [ x"$TERM" != "xscreen" ] 
-  function iterm2_status
-    printf "\033]133;D;%s\007" $argv
-  end
+set -l fishversion (echo $FISH_VERSION | sed 's|\(.\)\.\(.\).*|\1\2|')
+if [ $fishversion -ge 22 ]
+  if [ x"$TERM" != "xscreen" ]
 
-  # Mark start of prompt
-  function iterm2_prompt_start
-    printf "\033]133;A\007"
-  end
+    function iterm2_status -e fish_postexec
+      printf "\033]133;D;%s\007" $status
```

Comment:
Since we define two postexec functions how can we be sure $status is still correct? What if the other one runs first, will it mess up $status?